### PR TITLE
Add release notes for 2.6.2

### PR DESCRIPTION
**Pull request by madkinsz**
[Rendered](https://github.com/PrefectHQ/prefect/blob/release-notes-2.6.2/RELEASE-NOTES.md)

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,4 +1,29 @@
 # Prefect Release Notes
+## Release 2.6.2
+
+### Enhancements
+- Add `CompressedSerializer` for compression of other result serializers — https://github.com/PrefectHQ/prefect/pull/7164
+- Add option to drop task or flow return values from memory — https://github.com/PrefectHQ/prefect/pull/7174
+- Add support for creating and reading notification policies from the client — https://github.com/PrefectHQ/prefect/pull/7154
+- Add API support for sorting of deployments — https://github.com/PrefectHQ/prefect/pull/7187
```
Comment (billpalombi) [inline RELEASE-NOTES.md:8]:
```suggestion
- Add API support for sorting deployments — https://github.com/PrefectHQ/prefect/pull/7187
```

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,4 +1,29 @@
...
+- Improve searching and sorting of flows and deployments in the UI —  https://github.com/PrefectHQ/prefect/pull/7160
+- Improve rrule parsing with support for compound rules and rrulesets  — https://github.com/PrefectHQ/prefect/pull/7165
```
[the remark this answers is not in the document]
Comment (madkinsz) [reply RELEASE-NOTES.md:10]: It's not technically an all-caps thing, perhaps RRule would be most correct. How about the full name though?

```suggestion
- Improve recurrence rule schedule parsing with support for compound rules  — https://github.com/PrefectHQ/prefect/pull/7165
```

[the remark this answers is not in the document]
Comment (billpalombi) [reply RELEASE-NOTES.md]: Agreed thats better

**Comment by tpdorsey** [comment]
Ah these are not critical edits if you already merged

**Comment by madkinsz** [comment]
Ah sorry bad timing :D 

Thanks for looking though.

I haven't been capitalizing page names in the past, should I be?

**Comment by tpdorsey** [comment]
We probably should match the UI, but it's fine as is

**Pull request merged**